server/datastore/mysql/app_configs.go

```diff
@@ -145,35 +145,41 @@ func (d *Datastore) VerifyEnrollSecret(secret string) (*fleet.EnrollSecret, erro
 }
 
 func (d *Datastore) ApplyEnrollSecrets(teamID *uint, secrets []*fleet.EnrollSecret) error {
-	err := d.withRetryTxx(func(tx *sqlx.Tx) error {
-		if teamID != nil {
-			sql := `DELETE FROM enroll_secrets WHERE team_id = ?`
-			if _, err := tx.Exec(sql, teamID); err != nil {
-				return errors.Wrap(err, "clear before insert")
-			}
-		} else {
-			sql := `DELETE FROM enroll_secrets WHERE team_id IS NULL`
-			if _, err := tx.Exec(sql); err != nil {
-				return errors.Wrap(err, "clear before insert")
-			}
+	return d.withRetryTxx(func(tx *sqlx.Tx) error {
+		return d.applyEnrollSecretsDB(tx, teamID, secrets)
+	})
+}
+
+func (d *Datastore) applyEnrollSecretsDB(exec sqlx.Execer, teamID *uint, secrets []*fleet.EnrollSecret) error {
+	if teamID != nil {
+		sql := `DELETE FROM enroll_secrets WHERE team_id = ?`
+		if _, err := exec.Exec(sql, teamID); err != nil {
+			return errors.Wrap(err, "clear before insert")
+		}
+	} else {
+		sql := `DELETE FROM enroll_secrets WHERE team_id IS NULL`
+		if _, err := exec.Exec(sql); err != nil {
+			return errors.Wrap(err, "clear before insert")
 		}
+	}
 
-		for _, secret := range secrets {
-			sql := `
+	for _, secret := range secrets {
+		sql := `
 				INSERT INTO enroll_secrets (secret, team_id)
 				VALUES ( ?, ? )
 			`
-			if _, err := tx.Exec(sql, secret.Secret, teamID); err != nil {
-				return errors.Wrap(err, "upsert secret")
-			}
+		if _, err := exec.Exec(sql, secret.Secret, teamID); err != nil {
+			return errors.Wrap(err, "upsert secret")
 		}
-		return nil
-	})
-
-	return err
+	}
+	return nil
 }
 
 func (d *Datastore) GetEnrollSecrets(teamID *uint) ([]*fleet.EnrollSecret, error) {
+	return getEnrollSecretsDB(d.reader, teamID)
+}
+
+func getEnrollSecretsDB(q sqlx.Queryer, teamID *uint) ([]*fleet.EnrollSecret, error) {
```

Comment:
Is the idea to start having some kind of pattern `somethingDB`?

Reply:
Yeah, when there was a `Something` top-level method, but then somewhere else called it and needed to use the primary instead of the read, I extracted out the `Something` into an unexported `somethingDB(q Queryer, ...)` as some sort of naming convention to indicate that it receives the db interface as argument.

Happy to change it to something clearer if you have a suggestion, one thing I've seen elsewhere is a `Q` suffix to indicate it receives a `Queryer`, and a `Tx` for a transaction, though I'm not sure the difference matters - the interface used for the argument is the ~smallest that allows the function to do its job.

Reply:
I think that's ok. I wonder if all the methods that take the first argument as some kind of db interface should follow the same pattern.

Reply:
It's true that I started using this naming just for those that existed without it first - I'll try to update them all to this convention, I agree that it would make more sense.